bookstore/publish.py

```diff
@@ -0,0 +1,80 @@
+import json
+
+import aiobotocore
+
+from notebook.base.handlers import APIHandler
+from notebook.base.handlers import path_regex
+from notebook.utils import url_path_join
+from tornado import web
+
+from .s3_paths import s3_path
+from .s3_paths import s3_key
+from .s3_paths import s3_display_path
+from .bookstore_config import BookstoreSettings
+
+
+class BookstorePublishHandler(APIHandler):
+    """Publish a notebook to the publish path"""
+
+    def initialize(self):
+        """Initialize a helper to get bookstore settings and session information quickly"""
+        self.bookstore_settings = BookstoreSettings(config=self.config)
+        self.session = aiobotocore.get_session()
+
+    @web.authenticated
+    async def put(self, path=''):
+        """Publish a notebook on a given path.
+
+        The payload directly matches the contents API for PUT.
+        """
+        self.log.info("Attempt publishing to %s", path)
+
+        if path == '' or path == '/':
+            raise web.HTTPError(400, "Must provide a path for publishing")
+
+        model = self.get_json_body()
+        if model:
+            await self._publish(model, path.lstrip('/'))
+        else:
+            raise web.HTTPError(400, "Cannot publish an empty model")
+
+    async def _publish(self, model, path):
+        """Publish notebook model to the path"""
+        if model['type'] != 'notebook':
+            raise web.HTTPError(400, "bookstore only publishes notebooks")
+        content = model['content']
```

Comment:
I would skip this assignment and use `model['content']` when doing the put later.